Wikipedia/Code/WKWebView+WMFWebViewControllerJavascript.swift

```diff
@@ -0,0 +1,155 @@
+            return false
+        default:
+            break
+        }
+        return true
+    }
+    
+    public func itemAdditionJavascriptString(with article: MWKArticle) -> String {
+        var title = article.apostropheEscapedArticleLanguageLocalizedStringForKey(self.localizedTitleKey)
+        if let substitutionString = titleSubstitutionStringForArticle(article: article) {
+            title = title.replacingOccurrences(of: "$1", with: substitutionString)
+        }
+        
+        var subtitle = ""
+        if let subtitleKey = self.localizedSubtitleKey{
+            subtitle = article.apostropheEscapedArticleLanguageLocalizedStringForKey(subtitleKey)
+        }
+        
+        let itemSelectionHandler =
+        "function(){" +
+            "window.webkit.messageHandlers.footerMenuItemClicked.postMessage('\(footerMenuJSTransformEnumString)');" +
+        "}"
+        
+        return "window.wmf.footerMenu.addItem('\(title)', '\(subtitle)', \(self.footerMenuTransformJSEnumPath), 'footer_container_menu', \(itemSelectionHandler));"
+    }
+}
+
+extension WKWebView {
+    
+    public func wmf_addFooterMenuForArticle(_ article: MWKArticle){
+        let heading = article.apostropheEscapedArticleLanguageLocalizedStringForKey("article-about-title").uppercased(with: Locale.current)
+        evaluateJavaScript("window.wmf.footerMenu.setHeading('\(heading)', 'footer_container_menu_heading');", completionHandler: nil)
+
+        let itemsJS = [
+            WMFArticleFooterMenuItem.languages,
+            WMFArticleFooterMenuItem.coordinate,
+            WMFArticleFooterMenuItem.lastEdited,
+            WMFArticleFooterMenuItem.pageIssues,
+            WMFArticleFooterMenuItem.disambiguation
+            ].filter{$0.shouldAddItem(with: article)}
+             .map{$0.itemAdditionJavascriptString(with: article)}
+             .joined(separator: "")
+        
+        evaluateJavaScript(itemsJS, completionHandler: nil)
+    }
+
+    public func wmf_addFooterLegalForArticle(_ article: MWKArticle){
+        let licenseString = article.apostropheEscapedArticleLanguageLocalizedStringForKey("license-footer-text")
+        let licenseSubstitutionString = article.apostropheEscapedArticleLanguageLocalizedStringForKey("license-footer-name")
+        let licenseLinkClickHandler =
+        "function(){" +
+            "window.webkit.messageHandlers.footerLegalLicenseLinkClicked.postMessage('linkClicked');" +
+        "}"
+        evaluateJavaScript("window.wmf.footerLegal.add('\(licenseString)', '\(licenseSubstitutionString)', 'footer_container_legal', \(licenseLinkClickHandler));", completionHandler: nil)
+    }
+
+    public func wmf_addFooterReadMoreForArticle(_ article: MWKArticle){
+        guard
+            let proxyURL = WMFProxyServer.shared().proxyURL(forWikipediaAPIHost: article.url.host),
+            let title = (article.url as NSURL).wmf_title
+        else {
+            assert(false, "Expected read more title and proxyURL")
```

Comment:
more of an FYI, but instead of `assert(false, …` you can do

`assertionFailure("Expected read more title and proxyURL")`